experimental/packages/sdk-logs/src/LogRecordProcessor.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LogRecord } from './LogRecord';
+
+export interface LogRecordProcessor {
+  /**
+   * Forces to export all finished log records
+   */
+  forceFlush(): Promise<void>;
+
+  /**
+   * Called when a {@link LogRecord} is emit
+   * @param logRecord the ReadWriteLogRecord that just emitted.
+   */
+  onEmit(logRecord: LogRecord): void;
```

Comment:
I think we might need to add a Context parameter to this method ([see spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#onemit)), although I want to verify the intent here. It is not clear to me where exactly the span context should be added and if the Context here would be intended for the automatic span context injection.

Reply:
So yes, the Context parameter should be added to this method.

Also, the Logger should populate the trace context fields (spanId, traceId, traceFlags) when the `include_trace_context` is true (see [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/bridge-api.md#get-a-logger) and [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/bridge-api.md#implicit-context-injection)).

Reply:
I have noticed this problem. This is a new specification created during the duration of this pr. I think we should not support it in this pr right now,, because it will make this pr keep expanding, I will commit a new pr support after this pr merge

Reply:
fixed: feat(sdk-logs): logger option support includeTraceContext & LogRecordProcessor onEmit suport context https://github.com/open-telemetry/opentelemetry-js/pull/3549/commits/4cca19f753dbf5c2d7ad24012ecc13cd86a1565b